runtime/jvmti/jvmtiHook.c

```diff
@@ -1071,6 +1070,16 @@ jvmtiHookFieldModification(J9HookInterface** hook, UDATA eventNum, void* eventDa
 				if (methodID != NULL) {
 					callback((jvmtiEnv *) j9env, (JNIEnv *) currentThread, threadRef, methodID, (jlocation) location, (jclass) classRef, (jobject) objectRef, fieldID, signatureType, newValue);
 				}
+				vm->internalVMFunctions->internalEnterVMFromJNI(currentThread);
+				if (NULL != object) {
+					J9VMPutFieldEvent* data = eventData;
+					data->object = *objectRef;
+				}
+				if (signatureType == 'L') {
+					if (*((void **) valueAddress) != NULL) {
+						*(j9object_t*)valueAddress = *jvalueStorage;
```

Comment:
Why update `data->newValue` here?  `data` doesn't appear to be used after this point

Reply:
data is the hook event data, which may be used by another agent.